bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/profile/ProfileTypeResource.java

```diff
@@ -0,0 +1,98 @@
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.smarthome.core.auth.Role;
+import org.eclipse.smarthome.core.thing.profiles.ProfileType;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeRegistry;
+import org.eclipse.smarthome.core.thing.profiles.dto.ProfileTypeDTO;
+import org.eclipse.smarthome.core.thing.profiles.dto.ProfileTypeDTOMapper;
+import org.eclipse.smarthome.io.rest.LocaleUtil;
+import org.eclipse.smarthome.io.rest.RESTResource;
+import org.eclipse.smarthome.io.rest.Stream2JSONInputStream;
+import org.eclipse.smarthome.io.rest.core.internal.thing.ThingTypeResource;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * REST resource to obtain profile-types
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@Path(ProfileTypeResource.PATH_PROFILE_TYPES)
+@RolesAllowed({ Role.ADMIN })
+@Api(value = ProfileTypeResource.PATH_PROFILE_TYPES)
+@Component(service = { RESTResource.class, ProfileTypeResource.class })
+public class ProfileTypeResource implements RESTResource {
+
+    /** The URI path to this resource */
+    public static final String PATH_PROFILE_TYPES = "profile-types";
+
+    private final Logger logger = LoggerFactory.getLogger(ThingTypeResource.class);
+
+    private ProfileTypeRegistry profileTypeRegistry;
+
+    @GET
+    @RolesAllowed({ Role.USER })
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets all available profile types.", response = ProfileTypeDTO.class, responseContainer = "Set")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK", response = ProfileTypeDTO.class, responseContainer = "Set"))
+    public Response getAll(
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = HttpHeaders.ACCEPT_LANGUAGE) String language) {
+        Locale locale = LocaleUtil.getLocale(language);
+        Stream<ProfileTypeDTO> typeStream = profileTypeRegistry.getProfileTypes(locale).stream()
+                .map(t -> convertToProfileTypeDTO(t, locale));
+        return Response.ok(new Stream2JSONInputStream(typeStream)).build();
+    }
+
+    private ProfileTypeDTO convertToProfileTypeDTO(ProfileType profileType, Locale locale) {
+        final ProfileTypeDTO profileTypeDTO = ProfileTypeDTOMapper.map(profileType, locale);
+        if (profileTypeDTO != null) {
+            return profileTypeDTO;
+        } else {
+            logger.warn("Cannot create DTO for profileType '{}'. Skipping it.", profileTypeDTO);
+        }
+
+        return null;
+    }
+
+    @Reference
```

Comment:
Please annotate with `@Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)` to decouple the resource lifecycle from the OSGi service lifecycle.
Also: implement the `isSatisfied` method and check if the profileTypeRegistry was set.